File: src/main/java/net/minheur/potoflux/login/response/InfoResponse.java
package net.minheur.potoflux.login.response;

import com.google.gson.annotations.SerializedName;

/**
 * Info response class.<br>
 * It gives info on an account
 */
public class InfoResponse extends BaseResponse {
    /**
     * Accounts UUID
     */
    public String uuid;

    /**
     * Account's first name.
     */
    @SerializedName("first_name")
    public String firstName;

    /**
     * Account's last name.
     */
    @SerializedName("last_name")
    public String lastName;

    /**
     * Account's email.
     */
    @SerializedName("mail")
    public String email;

    /**
     * Account's perms.
     */
    public String[] perms;
    /**
     * Account's rank.
     */
    public int rank;
    /**
     * Weather account is locked.
     */
    public boolean locked;
}

File: src/main/kotlin/net/minheur/potoflux/login/response/InfoResponse.kt
package net.minheur.potoflux.login.response

import com.google.gson.annotations.SerializedName

class InfoResponse : BaseResponse() {
    @JvmField
    var uuid: String? = null

    @JvmField
    @SerializedName("first_name")
    var firstName: String? = null

    @JvmField
    @SerializedName("last_name")
    var lastName: String? = null

    @JvmField
    @SerializedName("mail")
    var email: String? = null

    @JvmField
    var perms: Array<String>? = null

    @JvmField
    var rank: Int = 0

    @JvmField
    var locked: Boolean = false
}
